stripped/colabs/mapy_datashader_stripped.py

```diff
@@ -0,0 +1,138 @@
+drugi argument funkcji.
+'''
+def get_colormap(values, colors_palette, name = 'custom'):
+    values = np.sort(np.array(values))
+    values = np.interp(values, (values.min(), values.max()), (0, 1))
+    cmap = mpl.colors.LinearSegmentedColormap.from_list(name, list(zip(values, colors_palette)))
+    return cmap
+
+
+"""
+Tworzymy pierwszą mapę, w tym wypadku dla parametru GVEG związanego z poziomem roślinności:
+"""
+
+colormap_GVEG = get_colormap([0, max(df.GVEG.values)], ['yellow', 'darkgreen'])
+
+gdf_GVEG = gv.Points(df, ['lon', 'lat'], ['GVEG']) # obiekt zawierający punkty
+
+tiles_GVEG = gts.OSM # wybieramy mapę tła, w tym wypadku OpenStreetMap
+
+# łączymy mapę tła z punktami i ustawiamy wybrane parametry wizualizacji
+map_with_points_gts_GVEG = tiles_GVEG * gdf_GVEG.opts(
+    color = 'GVEG',
+    cmap = colormap_GVEG,
+    size = 8,
+    width = 900,
+    height = 600,
+    colorbar = True,
+    toolbar = 'above',
+    tools = ['hover', 'wheel_zoom', 'reset'],
+    alpha = 0.5 # przezroczystość
+    )
+
+map_with_points_gts_GVEG.opts(bgcolor='white')
+
+# zapisujemy mapę do pliku .html
+hv.save(map_with_points_gts_GVEG.opts(width=900, height=600, bgcolor='white', xaxis=None, yaxis=None), 'output_map_osm_GVEG.html')
+#hv.save(map_with_points_gts_GVEG.opts(width=900, height=600, xaxis=None, yaxis=None), 'output_map_osm_GVEG_3.html')
+
+
+"""
+Druga mapa, tym razem dla parametru Rainf określającego poziom opadów deszczu:
+"""
+
+colormap_Rainf = get_colormap([0, max(df.Rainf.values)/2, max(df.Rainf.values)], ['white', 'blue', 'black'])
+
+gdf_Rainf = gv.Points(df, ['lon', 'lat'], ['Rainf'])
+
+tiles_Rainf = gts.OSM
+
+map_with_points_gts_Rainf = tiles_Rainf * gdf_Rainf.opts(
+    color = 'Rainf',
+    cmap = colormap_Rainf,
+    size = 5,
+    width = 900,
+    height = 600,
+    colorbar = True,
+    toolbar = 'above',
+    tools = ['hover', 'wheel_zoom', 'reset'],
+#    alpha = 0.75
+    )
+
+map_with_points_gts_Rainf.opts() #bgcolor='black')
+
+# hv.save(map_with_points_gts_Rainf.opts(width=900, height=600, bgcolor='black', xaxis=None, yaxis=None), 'output_map_osm_Rainf.html')
+hv.save(map_with_points_gts_Rainf.opts(width=900, height=600, xaxis=None, yaxis=None), 'output_map_osm_Rainf.html')
+
+
+"""
+Trzecia mapa dla AvgSurfT, tj. średniej temperatury powierzchni:
+"""
+
+colormap_AvgSurfT = 'cet_rainbow_bgyr_10_90_c83'
+
+gdf_AvgSurfT = gv.Points(df, ['lon', 'lat'], ['AvgSurfT'])
+
+tiles_AvgSurfT = gts.OSM
+
+map_with_points_gts_AvgSurfT = tiles_AvgSurfT * gdf_AvgSurfT.opts(
+    color = 'AvgSurfT',
+    cmap = colormap_AvgSurfT
+    size = 5,
+    width = 900,
+    height = 600,
+    colorbar = True,
+    toolbar = 'above',
+    tools = ['hover', 'wheel_zoom', 'reset'],
+    alpha = 0.7 # przezroczystość
+    )
+
+map_with_points_gts_AvgSurfT.opts(bgcolor='white')
+
+hv.save(map_with_points_gts_AvgSurfT.opts(width=900, height=600, bgcolor='white', xaxis=None, yaxis=None), 'output_map_osm_AvgSurfT.html')
```

Comment:
ciekawi mnie co przyjmuje hv.save za parametr. Co to za struktura. Bo wydaje się, że ploty na ekran powinny być łatwo wspierane [https://holoviews.org/getting_started/Introduction.html](https://holoviews.org/getting_started/Introduction.html) według choćby schematu:
```
scatter = hv.Scatter(station_info, 'services', 'ridership')
scatter
```